File: C#/BA1L.cs
using System;

namespace BA1L
{
    class Program
    {
        //A solution to a Rosalind bioinformatics problem
        //Problem Title: Implement PatternToNumber
        //URL: http://rosalind.info/problems/ba1l/
        static void Main(string[] args)
        {
            long PatternToNumber(string pattern)
            {
                long res = 0;
                for (int k = 0; k < pattern.Length; k++)
                {
                    if (pattern[pattern.Length-k-1] == 'C')
                    {
                        res = res + 1 * (long)(Math.Pow(4, k));
                    }
                    if (pattern[pattern.Length - k - 1] == 'G')
                    {
                        res = res + 2 * (long)(Math.Pow(4, k));
                    }
                    if (pattern[pattern.Length - k - 1] == 'T')
                    {
                        res = res + 3 * (long)(Math.Pow(4, k));
                    }
                }
                return res;
            }

            string x = "CTTCTCACGTACAACAAAATC";
            string[] inlines = x.Split();
            string text = inlines[0];
            Console.WriteLine(PatternToNumber(text));
        }
    }
}

File: Python/Chapter1/BA1L.py
#!/usr/bin/env python3

"""
A solution to a Rosalind bioinformatics problem
Problem Title: Implement PatternToNumber
URL: http://rosalind.info/problems/ba1l/
"""


def PatternToNumber(pattern):
    res = 0
    k = 0
    for x in pattern[::-1]:
        if x == "C":
            res = res + 1 * (4**k)
        if x == "G":
            res = res + 2 * (4**k)
        if x == "T":
            res = res + 3 * (4**k)
        k = k + 1
    return(res)


if __name__ == '__main__':

    x='''AGT'''
    inlines=x.split()
    pattern = inlines[0]

    res = PatternToNumber(pattern)
    print(str(res))

    x='''CTTCTCACGTACAACAAAATC'''
    inlines=x.split()
    pattern = inlines[0]

    res = PatternToNumber(pattern)
    print(str(res))

    x='''AACGCTGTGATCCGCACAGTCACGTCACGG'''
    inlines=x.split()
    pattern = inlines[0]

    res = PatternToNumber(pattern)
    print(str(res))
